common/src/models/domain/organization.ts

```diff
@@ -108,12 +108,32 @@ export class Organization {
         return this.isAdmin || this.permissions.accessReports;
     }
 
-    get canManageAllCollections() {
-        return this.isAdmin || this.permissions.manageAllCollections;
+    get canCreateNewCollections() {
+        return this.isAdmin || (this.permissions.createNewCollections ?? this.permissions.manageAllCollections);
     }
 
-    get canManageAssignedCollections() {
-        return this.isManager || this.permissions.manageAssignedCollections;
+    get canEditAnyCollection() {
+        return this.isAdmin || (this.permissions.editAnyCollection ?? this.permissions.manageAllCollections);
+    }
+
+    get canDeleteAnyCollection() {
+        return this.isAdmin || (this.permissions.deleteAnyCollection ?? this.permissions.manageAllCollections);
+    }
+
+    get canViewAllCollections() {
+        return this.canEditAnyCollection || this.canDeleteAnyCollection;
+    }
+
+    get canEditAssignedCollections() {
+        return this.isManager || (this.permissions.deleteAssignedCollections ?? this.permissions.manageAssignedCollections);
+    }
+
+    get canDeleteAssignedCollections() {
+        return this.isManager || (this.permissions.deleteAssignedCollections ?? this.permissions.manageAssignedCollections);
+    }
+
+    get canViewAssignedCollections() {
```

Comment:
Just a thought I had, would we gain anything from refactoring this to:

```
get canViewCollection(collection: Collection) {
	if (canEditAnyCollection || this.canDeleteAnyCollection) {
        return true;
    }
    if (collection.assigned == user) {
        return this.canDeleteAssignedCollections || this.canEditAssignedCollections;
    }
}
```

This might not quite be possible to do in the `Organization` domain model though and might belong in a permissions service or something similar.

Reply:
I think that's definitely a good idea, but I want to do it in a permissions service.

In general, both on clients and server, I want to modify the way we handle permissions from asking for general permission to specific actions on specific models. What we're doing now uses knowledge of the guid id as implicit read permission, which I don't like.